api/v1alpha3/azuremachine_webhook.go

```diff
@@ -87,6 +89,10 @@ func (m *AzureMachine) ValidateUpdate(old runtime.Object) error {
 		allErrs = append(allErrs, errs...)
 	}
 
+	if errs := ValidateManagedDisk(old.Spec.OSDisk.ManagedDisk, m.Spec.OSDisk.ManagedDisk, field.NewPath("osDisk").Child("managedDisk")); len(errs) > 0 {
```

Comment:
Can this instead call `ValidateOSDisk` which itself calls `ValidateManagedDisk`? I don't see anything in `ValidateOSDisk` that applies to AzureCluster Create but not Update

Reply:
diffDiskSettings can only be set on OS disks, I need to pull that up a layer. That's the difference between ValidateOSDisk and ValidateManagedDisk

Reply:
actually this field can't be set for managed disks which aren't os disks, so I'll also validate against that.

Reply:
*taps head* don't need to validate if it's not in the API

i'll move it up a level to osDisk struct